web/server/codechecker_server/api/report_server.py

```diff
@@ -3424,10 +3424,24 @@ def removeRun(self, run_id, run_filter):
             # cascades and such. Deleting runs in separate transactions don't
             # exceed a potential statement timeout threshold in a DBMS.
             runs = []
+            run_no = 0
+
             for run in q.all():
-                runs.append(run.name)
-                session.delete(run)
-                session.commit()
+                try:
+                    runs.append(run.name)
+                    session.delete(run)
+                    session.commit()
+                    run_no += 1
+                except Exception as e:
+                    # TODO: Display alert on the GUI if there's an exception
+                    # This exception is handled silently because it is
+                    # expected to never occur, but there have been some rare
+                    # cases where it occurred due to underlying reasons.
+                    # Handling it silently ensures that the Number of runs
+                    # counter is not affected by the exception.
```

Comment:
One reason of the exception can be an SQL statement timeout while a too large run is being deleted. We should include another TODO here which suggests the catching of `SQLAlchemyError` instead of the generic `Exception` when we can make this sure based on the warnings messages below, in the server logs.